headers/components/temp_doeclim_component.hpp

```diff
@@ -0,0 +1,150 @@
+/* Hector -- A Simple Climate Model
+   Copyright (C) 2014-2015  Battelle Memorial Institute
+
+   Please see the accompanying file LICENSE.md for additional licensing
+   information.
+*/
+#ifndef TEMP_DOECLIM_COMPONENT_H
+#define TEMP_DOECLIM_COMPONENT_H
+/*
+ *  temp_doeclim_component.hpp
+ *  hector
+ *
+ *  Created by BenV-W on 11/01/2016.
+ *
+ */
+
+#include "components/forcing_component.hpp"
+#include "components/imodel_component.hpp"
+#include "core/logger.hpp"
+#include "data/tseries.hpp"
+#include "data/unitval.hpp"
+
+
+namespace Hector {
+  
+//------------------------------------------------------------------------------
+/*! \brief Temperature component.
+ *
+ *  A component that computes mean global temperature from radiative forcing 
+ *  using Diffusion Ocean Energy balance CLImate Model (DOECLIM), (Kriegler, 2005).
+ *  Adopted from C++ github implementation, cdice_doeclim (Garner et al., 2013).
+ */
+class TempDOECLIMComponent : public IModelComponent {
+    
+public:
+    TempDOECLIMComponent();
+    ~TempDOECLIMComponent();
+    
+    //! IModelComponent methods
+    virtual std::string getComponentName() const;
+    
+    virtual void init( Core* core );
+    
+    virtual unitval sendMessage( const std::string& message,
+                                const std::string& datum,
+                                const message_data info=message_data() ) throw ( h_exception );
+    
+    virtual void setData( const std::string& varName,
+                          const message_data& data ) throw ( h_exception );
+    
+    virtual void prepareToRun() throw ( h_exception );
+    
+    virtual void run( const double runToDate ) throw ( h_exception );
+    
+    virtual void shutDown();
+    
+    //! IVisitable methods
+    virtual void accept( AVisitor* visitor );
+    
+private:
+    virtual unitval getData( const std::string& varName,
+                            const double date ) throw ( h_exception );
+    void invert_1d_2x2_matrix( double * x, double * y);
+    void sum_1d_2x2_matrix( double * x, double * y, double * z);
+    
+    //Hard-coded DOECLIM parameters
+    int dt;                  // years per timestep (this is hard-coded into Hector)
+    int ns;                  // number of timesteps
+    double ak;               // slope in climate feedback - land-sea heat exchange linear relationship
+    double bk;               // offset in climate feedback - land-sea heat exchange linear relationship, W/m2/K
+    double csw;              // specific heat capacity of seawater W*yr/m3/K
+    double earth_area;       // m2
```

Comment:
Seems like many of these should be `const`

Reply:
instead of `doubles`?

Reply:
No, I mean, something like `const double earth_area = x.xxx;`

Reply:
Based on @rplzzz 's comment below it looks like we should move these initializations to the header (some are called in more than one function so if we want to keep them together they need to be there). Those with well-known values will have `const static`, while those that are potentially tweakable will have `const`. Does that sound like what we want?